x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/expression/function/scalar/string/Chunk.java

```diff
@@ -0,0 +1,235 @@
+            throw new InvalidArgumentException("[{}] cannot be negative, found [{}]", NUM_CHUNKS, numChunks);
+        }
+        Integer chunkSize = (Integer) optionsMap.get(CHUNK_SIZE);
+        if (chunkSize != null && chunkSize < 0) {
+            throw new InvalidArgumentException("[{}] cannot be negative, found [{}]", CHUNK_SIZE, chunkSize);
+        }
+
+    }
+
+    @Override
+    public boolean foldable() {
+        return field().foldable() && (options() == null || options().foldable());
+    }
+
+    @Override
+    public Expression replaceChildren(List<Expression> newChildren) {
+        return new Chunk(
+            source(),
+            newChildren.get(0), // field
+            newChildren.size() > 1 ? newChildren.get(1) : null // options
+        );
+    }
+
+    @Override
+    protected NodeInfo<? extends Expression> info() {
+        return NodeInfo.create(this, Chunk::new, field, options);
+    }
+
+    Expression field() {
+        return field;
+    }
+
+    Expression options() {
+        return options;
+    }
+
+    @Evaluator(extraName = "BytesRef")
+    static void process(BytesRefBlock.Builder builder, BytesRef str, int numChunks, int chunkSize) {
+        String content = str.utf8ToString();
+
+        ChunkingSettings settings = new SentenceBoundaryChunkingSettings(chunkSize, 0);
+        List<String> chunks = chunkText(content, settings, numChunks);
+
+        boolean multivalued = chunks.size() > 1;
+        if (multivalued) {
+            builder.beginPositionEntry();
+        }
+        for (String chunk : chunks) {
+            builder.appendBytesRef(new BytesRef(chunk.trim()));
+        }
+
+        if (multivalued) {
+            builder.endPositionEntry();
+        }
+    }
+
+    public static List<String> chunkText(String content, ChunkingSettings chunkingSettings, int numChunks) {
+        Chunker chunker = ChunkerBuilder.fromChunkingStrategy(chunkingSettings.getChunkingStrategy());
+        return chunker.chunk(content, chunkingSettings)
+            .stream()
+            .map(offset -> content.substring(offset.start(), offset.end()))
+            .limit(numChunks > 0 ? numChunks : DEFAULT_NUM_CHUNKS)
+            .toList();
+    }
+
+    @Override
+    public boolean equals(Object o) {
+        if (o == null || getClass() != o.getClass()) return false;
+        Chunk chunk = (Chunk) o;
+        return Objects.equals(field(), chunk.field()) && Objects.equals(options(), chunk.options());
+    }
+
+    @Override
+    public int hashCode() {
+        return Objects.hash(field(), options());
+    }
+
+    @Override
+    public EvalOperator.ExpressionEvaluator.Factory toEvaluator(ToEvaluator toEvaluator) {
+
+        MapExpression optionsMap = options() != null ? (MapExpression) options() : null;
+
+        return new ChunkBytesRefEvaluator.Factory(
+            source(),
+            toEvaluator.apply(field),
+            optionsMap != null && optionsMap.containsKey(NUM_CHUNKS)
```

Comment:
Using options means that we have literal values - we can simplify this and use `Options.populateMap()` to get a `Map<String, Object>` instead of evaluating the options.